### ISSUE TITLE
Lots of accessibility issues with X64DBG

### ISSUE THREAD

**Issue by akash07k** [x64dbg/x64dbg#2234]
Dear developers and team,
First of all, I thank you all for this great tool.
It has really a great potential.
But the issue that the accessibility is majorly and badly lacking in X64DBG.
For example, there are many controls which are not having any labels such as edit ffields for typing the strings, lists containing the strings etc.
The primary issue I've faced that when we search for some strings, the list is not accessible at all and we are unable to change any string or JMP/JNE calls.
Please help.
since the application is using QT5 Widgets, please refer this link:
https://doc.qt.io/qt-5/accessible.html

**Comment by akash07k** [comment]
Please refer these links in order to know that what is accessibility and why is it important:
https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility
https://www.w3.org/WAI/fundamentals/accessibility-intro/
https://accessibility.blog.gov.uk/2016/05/16/what-we-mean-when-we-talk-about-accessibility-2/
https://www.usability.gov/what-and-why/accessibility.html

**Comment by mrexodia** [comment]
Please document specific examples you think are lacking. I’m usually not using the mouse too much and Qt should take care of most things automatically. Due to the nature of x64dbg it will never be compatible with screen readers, but the tab order and &menu item should allow you to use x64dbg.

**Comment by akash07k** [comment]
> Please document specific examples you think are lacking. I’m usually not using the mouse too much and Qt should take care of most things automatically. Due to the nature of x64dbg it will never be compatible with screen readers, but the tab order and &menu item should allow you to use x64dbg.

As I said before, the list of the strings is completel  inaccessible with the keyboard and to the screen readers.

**Comment by akash07k** [comment]
> Please document specific examples you think are lacking. I’m usually not using the mouse too much and Qt should take care of most things automatically. Due to the nature of x64dbg it will never be compatible with screen readers, but the tab order and &menu item should allow you to use x64dbg.

Also, the edit fields for entering the command, searching the strings are not having any labels.
We are not even able to change the JMP, JNE etc calls. that part is also completely inaccessible

**Comment by akash07k** [comment]
> Please document specific examples you think are lacking. I’m usually not using the mouse too much and Qt should take care of most things automatically. Due to the nature of x64dbg it will never be compatible with screen readers, but the tab order and &menu item should allow you to use x64dbg.

The data/UI under each tab such as CPU, GRAPH, Call Stack, Sources etc is also not accessible at all.

**Comment by akash07k** [comment]
@mrexodia  Any update on this please?
I hope accessibility can be considered so that screen reader users can also use this tool for their needs.
🙏

**Comment by mrexodia** [comment]
Unfortunately this is not something I plan to work on. Due to the nature of the custom controls it would require 100% custom screen reader support and I don’t use one so this is not something I’ll be working on.

**Comment by akash07k** [comment]
That's very sad. 😔
> Unfortunately this is not something I plan to work on. Due to the nature of the custom controls it would require 100% custom screen reader support and I don’t use one so this is not something I’ll be working on.

**Comment by torusrxxx** [comment]
x64dbg is based on a few basic custom widgets, most importantly table. There is also a built-in function to allow you to copy table content, which could be read by a screen reader. However I don't think x64dbg would be usable by using screen reader only. You have to listen to lots of information and it is very confusing, almost impossible to comprehend.

**Comment by torusrxxx** [comment]
I have added text labels in #3152. Making x64dbg completely accessible is hard. A Qt plugin library has to be created as recommended by https://doc.qt.io/qt-5/accessible.html. I turned on a screenreader, but I noticed even Qt Creator itself is barely usable with screenreader. I will spend reasonable effort in improving accessibility, but there are other priorities, for example, tab stops are currently a mess.

**Comment by akash07k** [comment]
Thanks for adding the text labels @torusrxxx 
Yes, making QT based applications accessible is a tricky affair.
I appreciate all your efforts which you are going to put for improving the accessibility.

> I have added text labels in #3152. Making x64dbg completely accessible is hard. A Qt plugin library has to be created as recommended by https://doc.qt.io/qt-5/accessible.html. I turned on a screenreader, but I noticed even Qt Creator itself is barely usable with screenreader. I will spend reasonable effort in improving accessibility, but there are other priorities, for example, tab stops are currently a mess.

**Comment by torusrxxx** [comment]
Accessibility will be a future goal of x64dbg GUI. Next time open a separate report for each feature request related to accessibility.

**Comment by akash07k** [comment]
@torusrxxx  ok
